### PR TITLE
Connect Kusama to live Circuit via XDNS - Closes #122

### LINKED ISSUE #122: Connect Kusama to live Circuit via XDNS
### Description

XDNS serves metadata with registry of up-to-date Chains.

Use entry register_gateway (available_methods: "transfer, get_storage", vendor: "Substrate" ) on ExecDelivery to register Kusama and cover with RPC tests. As a result we want to fetch metadata about Kusama to be exposed for SDK. 

!Remember! - Connect to archiving nodes only to make sure u have the info from below 250 blocks back 

### Acceptance 

- [x] Subtask - extend register_gateway in ExecDelivery to accept the parameters needed in instantiate Chain 
- [ ] Subtask - manually / in test file Test RPC transfer + get_storage at Westend and get_storage on Kusama (Test RPC transfer + get_storage inside the implementation when "on-register")
- [x] Add XDNS RPC to Circuit RPC
- [x] Implement runtime API for XDNS
- [x] Add check for sudo origin on register_gateway function
- [x] Test RPC endpoint for success and failure
- [x] stub for the further us (done as part of #160 )
- [x] Subtask - check if register_gateway is clicking with pallet-multi-verifier bridge that starts and running
- [x] Subtask - test that pallet-bridge-multi .. returns the latest headers for that gateways (that bridge

### PR DESCRIPTION
**Pull request by zannis**
Draft PR that closes #122 

- [x] Implement RPC endpoints for pallet XDNS + Contracts Registry
- [x] Expose them on Circuit's API
- [x] Test `register_gateway` on live circuit
- [x] Fixes broken fixtures symlink on Gateway that broke CI
- [x] Restores Circuit Release CI running only on merges to development branch
- [x] This PR also includes a significant amount of work on Relayer for #147 but to be continued on a future branch
- [x] Removes `signed_extension` from GatewayGenesisConfig as it was unused and was causing issues
- [x] Some integration tests are ignored until event emitting functionality is implemented

### CODE REVIEW

**Comment by Bongulielmi** [comment]
Task linked: [REG-116 Connect Kusama to live Circuit via XDNS](https://app.clickup.com/t/24316129/REG-116)

**Comment by Bongulielmi** [comment]
Task linked: [3RN-99 4 connected live integrations](https://app.clickup.com/t/24316129/3RN-99)

**Review by MaciejBaj — CHANGES_REQUESTED** [review]
Please add short README.md on starting the relayer client

File: circuit/pallets/execution-delivery/src/benchmarking.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ fn insert_default_xdns_record<T: pallet_xdns::Config>() {
 
     let gateway_genesis = GatewayGenesisConfig {
         modules_encoded: None,
-        signed_extension: None,
+        // signed_extensions: None,
```
Comment (MaciejBaj) [inline circuit/pallets/execution-delivery/src/benchmarking.rs:164]: why commented out?

File: circuit/pallets/execution-delivery/src/exec_composer/tests.rs [hunk 2]
```diff
@@ -110,7 +110,7 @@ pub fn make_compose_out_of_raw_wat_code<T: Config>(
 }
 
 pub fn insert_default_xdns_record() {
-    use pallet_xdns::XdnsRecord;
+    use pallet_xdns::types::XdnsRecord;
```
Comment (MaciejBaj) [inline circuit/pallets/execution-delivery/src/exec_composer/tests.rs:113]: Types vs t3rn-primitives crate? @zannis let's leave in types for now but please extract as a separate issue to align it.

**Review by MaciejBaj — CHANGES_REQUESTED** [review]

File: circuit/pallets/execution-delivery/src/exec_composer/tests.rs (hunk 2 above)
Comment (zannis) [reply circuit/pallets/execution-delivery/src/exec_composer/tests.rs:113]: Ah I see, yeah I just moved the types to a separate file for clarity, but I can re-export them on the lib.rs so this is aligned with the rest of the codebase.

File: circuit/pallets/execution-delivery/src/benchmarking.rs (hunk 1 above)
Comment (zannis) [reply circuit/pallets/execution-delivery/src/benchmarking.rs:164]: This one was breaking the decoding of payloads from js clients and is also unused in the codebase, so I commented it out as a workaround until we need it for some other implementation.

**Comment by zannis** [comment]
> Please add short README.md on starting the relayer client

@devanshu0987

**Review by MaciejBaj — APPROVED** [review]

**Pull request merged**